webhookValidationConfig_test.go

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright 2021 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package ancla
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	mockNow = func() time.Time {
+		return time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC)
+	}
+	mockMax        = 5 * time.Minute
+	mockJitter     = 5 * time.Second
+	buildAllConfig = ValidatorConfig{
+		URL: URLVConfig{
+			HTTPSOnly:            true,
+			AllowLoopback:        false,
+			AllowIP:              false,
+			AllowSpecialUseHosts: false,
+			AllowSpecialUseIPs:   false,
+			InvalidHosts:         []string{},
+			InvalidSubnets:       []string{},
+		},
+		TTL: TTLVConfig{
+			Max:    mockMax,
+			Jitter: mockJitter,
+			Now:    mockNow,
+		},
+	}
+	buildNoneConfig = ValidatorConfig{
+		URL: URLVConfig{
+			HTTPSOnly:            false,
+			AllowLoopback:        true,
+			AllowIP:              true,
+			AllowSpecialUseHosts: true,
+			AllowSpecialUseIPs:   true,
+			InvalidHosts:         []string{},
+			InvalidSubnets:       []string{},
+		},
+		TTL: TTLVConfig{
+			Max:    mockMax,
+			Jitter: mockJitter,
+			Now:    mockNow,
+		},
+	}
+)
+
+//just check if validators is nil or not and check errors
+//or check webhooks and see if the validators did what they were supposed to on them
+
+func TestBuildValidURLFuncs(t *testing.T) {
+	tcs := []struct {
+		desc            string
+		config          ValidatorConfig
+		expectedErr     error
+		expectedOutcome int
```

Comment:
```suggestion
		expectedFuncCount int
```
Outcome is kind of vague...this is my suggested name but something similar to `expectedNumOfFuncs` also works.